src/agents/extensions/models/litellm_model.py

```diff
@@ -367,15 +368,48 @@ async def _fetch_response(
         if isinstance(ret, litellm.types.utils.ModelResponse):
             return ret
 
+        # Convert tool_choice to the correct type for Response
```

Comment:
can you move these lines of logic into a separate static function and add some unit tests?